### ISSUE TITLE
Location in the background?

### ISSUE THREAD

**Issue by russmenum** [douglasjunior/react-native-get-location#22]
This is working fine when the app is in the foreground, but not in the background. 

For Android I have added: 
```
<uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
<uses-permission android:name="android.permission.ACCESS_BACKGROUND_LOCATION" />
```

Fr iOS:
```
        <key>NSLocationAlwaysAndWhenInUseUsageDescription</key>
	<string>Allow $(PRODUCT_NAME) to use your location</string>
	<key>NSLocationAlwaysUsageDescription</key>
	<string>Allow $(PRODUCT_NAME) to use your location</string>
	<key>NSLocationWhenInUseUsageDescription</key>
	<string>Allow $(PRODUCT_NAME) to use your location</string>
```

When trying to test this using ```react-native-background-actions``` I get some progress with iOS (in that I get an error), but nothing for Android. What is needed to make this run correctly as a background task

- How did you link to the library (autolinking, manual, rn link, cocoapods)?
RN 63 autolinking android
cocoapods pod install iOS

- What version of React Native?
```"react-native": "~0.63.4",```

- What version of the library?
- ```"react-native-get-location": "^2.1.1",```
- 
- iOS/Android version?
many

- Did the problem happen after updating React Native?
No, fresh ```npx create-react-native-app``` 
 
- Are you using the library for the first time?
Trying to...

**Comment by douglasjunior** [comment]
I've never tested using it in the background, I haven't done anything to prepare it either, so it's expected to have problems.

What errors are you getting?

I know that currently both iOS and Android have permissions that are only released when the app is in the Foreground, so this could be one of the problems.

**Comment by douglasjunior** [comment]
Even on iOS the permission used by Lib is `WhenInUseAuthorization`. 

https://github.com/douglasjunior/react-native-get-location/blob/df0f4523ca3f0b3e5a687b872febd42f2e51adee/ios/ReactNativeGetLocationLibrary/LocationModule.m#L67

You can try to get background permissions using another library, like "react-native-permissions".

**Comment by russmenum** [comment]
To clerify:

on iOS I was getting ```CANCELLED Location cancelled by another request``` but it seems to have stopped at least in the simulator after a rebuild and passing some prams by props to the background/foreground task.

But for Android, I get nothing, not even with the app in the foreground once I move 
```
GetLocation.getCurrentPosition({
                enableHighAccuracy: true,
                timeout: 15000,
              })
```
to the background HeadlessJS.

I am using [react-native-background-actions](https://github.com/Rapsssito/react-native-background-actions#readme)

and trying to run ```GetLocation.getCurrentPosition()``` and the POST its results in the ```veryIntensiveTask ``` like you see in this [exsample code](https://github.com/Rapsssito/react-native-background-actions/blob/master/USAGE.md)

Basically loop over your ```GetLocation.getCurrentPosition()``` every x sec. The way this seems to work is in the foreground when the app is in the background by making the task part of a notification that is always up on android using React Native's HeadlessJS for Android. I see the notification come up on my test phone, but seeing no activity from ```GetLocation.getCurrentPosition()``` I even set it up to POST if it gets an error but still nothing.

**Comment by russmenum** [comment]
OK, this may have nothing per say to do with this library or permissions; but I am not 100% sure. 

Long and short the [react-native-background-actions](https://github.com/Rapsssito/react-native-background-actions#readme) I am trying to use with react-native-get-location is supposed to open a Notification and run the JS you give it even if the app is not in the foreground because the notification is active. But how it does this for iOS and Android is different. Android needs the notification, but iOS handles this differently.

Upon more debugging; I found their ```await BackgroundService.start(veryIntensiveTask, options);``` is not actually starting the foreground notification that would be the task to run; in this case  ```GetLocation.getCurrentPosition()```. I was seeing a notification but looks like that was only the ```await BackgroundService.updateNotification()``` and so seems to not actually have the task to run.

So assuming it is that wont work on Android; which may or may not be the case. Is there one that will run well with ```GetLocation.getCurrentPosition()``` on Android? I am not 100% sure it is also not permissions; as React Native's HeadlessJS for Android as a general rule can not touch UX, it is a side separate JS thread. 

Basically, it needs to have all it needs before it ever gets to starting running there, or it wont work. If anything is needed it needs to be passed in by props to have it in scope when you try and start it up. 

I may just be overlooking something, so if you see anything obvious that would conflict in this use case I am all ears. KInd of at that no clue why point.
